tests/textcode/test_analysis.py

```diff
@@ -91,7 +96,8 @@ def test_numbered_text_lines_handles_sfdb(self):
         test_file = self.get_test_loc('analysis/splinefonts/Ambrosia.sfd')
         result = list(l for _, l in numbered_text_lines(test_file))
         expected_file = test_file + '.expected'
-        expected = open(expected_file, 'rb').read().splitlines(True)
+        mode = 'rb' if on_linux and py2 else 'r'
```

Comment:
The more should be the same on all OSses. These are not paths here but file contents.... so there is no difference between python2 and 3 and linux or not.

Note: The approach to read as binary and split lines keeping ends what there to ensure that we would keep the CRL/LF if any if I recall OK. This is not essential to keep line endings but it may be difficult to avoid reading bytes in. I am not sure we can get text in at first until we have eventually detected encoding and so on.